frontend/src/pages/ActivityReport/Pages/goalsObjectives.js

```diff
@@ -201,8 +203,9 @@ const GoalsObjectives = ({
     // so 'create a new goal' will still be an option.
   };
 
-  const onRemove = (goal) => {
-    const goalId = goal.id;
+  let goalToRemove = null;
```

Comment:
While this works now, I think it's risky not to store ```goalToRemove``` as a ref or in state

Reply:
updated to state.